app/src/main/kotlin/com/simplemobiletools/flashlight/activities/MainActivity.kt

```diff
@@ -1,332 +1,367 @@
-                config.stroboscopeProgress = progress
-            }
-        }
+        startAboutActivity(R.string.app_name, 0, BuildConfig.VERSION_NAME, faqItems, true)
     }
 
-    private fun toggleStroboscope(isSOS: Boolean) {
+    private fun toggleStroboscope(isSOS: Boolean, launcher: ManagedActivityResultLauncher<String, Boolean>) {
         // use the old Camera API for stroboscope, the new Camera Manager is way too slow
         if (isNougatPlus()) {
             cameraPermissionGranted(isSOS)
         } else {
-            handlePermission(PERMISSION_CAMERA) {
-                if (it) {
-                    cameraPermissionGranted(isSOS)
-                } else {
-                    toast(R.string.camera_permission)
-                }
+            val permission = Manifest.permission.CAMERA
+            if (ContextCompat.checkSelfPermission(this, permission) == PackageManager.PERMISSION_GRANTED) {
+                cameraPermissionGranted(isSOS)
+            } else {
+                launcher.launch(permission)
             }
         }
     }
 
-    private fun setupBrightness() {
-        binding.brightnessBar.apply {
-            max = mCameraImpl?.getMaximumBrightnessLevel() ?: MIN_BRIGHTNESS_LEVEL
-            progress = mCameraImpl?.getCurrentBrightnessLevel() ?: MIN_BRIGHTNESS_LEVEL
-            onSeekBarChangeListener { level ->
-                val newLevel = level.coerceAtLeast(MIN_BRIGHTNESS_LEVEL)
-                mCameraImpl?.updateBrightnessLevel(newLevel)
-                config.brightnessLevel = newLevel
-            }
+    private fun getPermissionResultHandler(isSos: Boolean): (Boolean) -> Unit = {
+        handlePermissionResult(isSos, it)
+    }
+
+    private fun handlePermissionResult(isSos: Boolean, granted: Boolean) {
+        if (granted) {
+            cameraPermissionGranted(isSos)
+        } else {
+            toast(R.string.camera_permission)
         }
     }
 
     private fun cameraPermissionGranted(isSOS: Boolean) {
         if (isSOS) {
-            val isSOSRunning = mCameraImpl!!.toggleSOS()
-            binding.sosBtn.setTextColor(if (isSOSRunning) getProperPrimaryColor() else getContrastColor())
-        } else if (mCameraImpl!!.toggleStroboscope()) {
-            binding.apply {
-                stroboscopeBar.beInvisibleIf(stroboscopeBar.isVisible())
-                changeIconColor(if (stroboscopeBar.isVisible()) getProperPrimaryColor() else getContrastColor(), stroboscopeBtn)
-            }
+            viewModel.enableSos()
+        } else {
+            viewModel.enableStroboscope()
         }
     }
 
-    private fun getContrastColor() = getProperBackgroundColor().getContrastColor()
-
-    private fun releaseCamera() {
-        mCameraImpl?.releaseCamera()
-        mCameraImpl = null
-    }
-
-    private fun showSleepTimer(force: Boolean = false) {
+    private fun showSleepTimerPermission(
+        showSleepTimerDialogState: AlertDialogState,
+        callback: () -> Unit
+    ) {
```

Comment:
don't pass an alert dialog state as a parameter, rather a callback

also make sure to use more descriptive name than callback